test/testdrive/temporary.td

```diff
@@ -67,6 +67,19 @@ catalog item 'foo' already exists
 
 > CREATE OR REPLACE TEMPORARY MATERIALIZED VIEW foo AS SELECT * FROM v
 
+> CREATE TEMPORARY TABLE temp_t (a int, b text NOT NULL)
```

Comment:
I am afraid GitHub does not allow global-yet-individually-resolveable conversations in the general Comments section, so I will try adding my stuff as individual comments against the test file instead.

Item #1. Temporary tables do not behave correctly if there is an actual table with the same name. The temporary table needs to mask the existing one while it is alive.

In this branch:
```
pstoev=> create table t1 (f1 integer);
CREATE TABLE
pstoev=> insert into t1 values (1);
INSERT 0 1
pstoev=> create temporary table t1 (f1 integer);
CREATE TABLE
pstoev=> insert into t1 values (2);
INSERT 0 1
pstoev=> select * from t1;
 f1 
----
  1
  2 <- wrong, the two inserts got mixed together at some point
(2 rows)

pstoev=> drop table t1;
DROP TABLE
pstoev=> select * from t1;
 f1 
----
(0 rows) <- wrong, the data from the non-temporary table should have shown up.
```

Contrast with Postgresql:

```
philips=> create table t1 (f1 integer);
CREATE TABLE
philips=> insert into t1 values (1);
INSERT 0 1
philips=> create temporary table t1 (f1 integer);
CREATE TABLE
philips=> insert into t1 values (2);
INSERT 0 1
philips=> select * from t1;
 f1 
----
  2 <- data from the temporary table is visible here
(1 row)

philips=> drop table t1;
DROP TABLE
philips=> select * from t1;
 f1 
----
  1 <- data from the original table is returned
```

Reply:
I didn't realize this!

> Existing permanent tables with the same name are not visible to the current session while the temporary table exists, unless they are referenced with schema-qualified names. Any indexes created on a temporary table are automatically temporary as well.

Since this is also true for views, this is also broken for our views:
```
materialize=> create view test as select 1;
CREATE VIEW
materialize=> create temp view test as select 2;
CREATE VIEW
materialize=> select * from test;
 ?column?
----------
        1
(1 row)
```

Reply:
Fixed this locally for all temporary objects!